backend/generate.ts

```diff
@@ -0,0 +1,185 @@
+import Instructor from "@instructor-ai/instructor";
+import OpenAI from "openai";
+import { z } from "zod";
+import readline from "readline";
+
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+});
+
+const client = Instructor({
+  client: oai,
+  mode: "TOOLS",
+});
+
+export const promptWithSchema = async (
```

Comment:
We should find a way to type this. I don't know exactly how Zod works, but generally you would use a generic type, eg:
```
async function promptWithScema<T>(prompt: string, schema: z.ZodObject<T>) { ... }
```
And then when you call the function, idk if TS can infer, but if not, you can call it with `promptWithSchema<UniverseSchema>` and it should get the appropriate return type. (Barring my lack of knowledge of Zod's typing.)